### PR TITLE
Add generics for info and storage commands

### PR DESCRIPTION
**Pull request by smiasojed**
## Summary
Related #1168
- [n] y/n | Does it introduce breaking changes?
- [n] y/n | Is it dependent on the specific version of `ink` or `pallet-contracts`?

Add generics for the 'info' and 'storage' commands

## Checklist before requesting a review
- [x] My code follows the style guidelines of this project
- [ ] I have added an entry to `CHANGELOG.md`
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: crates/extrinsics/src/contract_storage.rs [hunk 1]
```diff
@@ -75,21 +77,26 @@ use super::{
...
     ContractInfo,
-    DefaultConfig,
     TrieId,
 };
 
-pub struct ContractStorage<C: Config = DefaultConfig> {
-    rpc: ContractStorageRpc<C>,
+pub struct ContractStorage<C: Config, E: Environment> {
+    rpc: ContractStorageRpc<C, E>,
+    _phantom: PhantomData<fn() -> (C, E)>,
```
Comment (ascjones) [inline crates/extrinsics/src/contract_storage.rs:85]: Isn't this redundant because the type params are used in `ContractStorageRpc`?

**Review by ascjones — APPROVED** [review]
LGTM

File: crates/extrinsics/src/contract_storage.rs (hunk 1 above)
Comment (smiasojed) [reply crates/extrinsics/src/contract_storage.rs:85]: I will remove it. Thanks

**Review by SkymanOne — APPROVED** [review]
LGTM. Please add changelog entry before you merge

**Comment by smiasojed** [comment]
> LGTM. Please add changelog entry before you merge

I am not sure if I should add it into changelog, it does not affect an user, it is just internal change. WDYT?

**Comment by SkymanOne** [comment]
> > LGTM. Please add changelog entry before you merge
> 
> I am not sure if I should add it into changelog, it does not affect an user, it is just internal change. WDYT?

Generally speaking, unless the change is <10 loc, it's worth adding an entry just for internal reporting reasons so we know what and when it was changed.

**Review by cmichi — APPROVED** [review]
I'm fine with not having a changelog entry, as the change doesn't change behavior of `cargo-contract` for users.

**Pull request merged**